### DISCUSSION TITLE
Switching dGPU to iGPU in my laptop

### DISCUSSION THREAD

**Discussion by b1kr3m** [prasanthrangan/hyprdots#1242] [Q&A]
Currently in my laptop I'm using NVIDIA 4060 as a default GPU, but somehow it takes lots of battery and facing some heat issues. could you please help me, how i can change my dGPU to iGPU. last time i tried but i was accidentally broke my system. 😅 .
i am a new Arch User, so please guide me. 
this is my system info:
                                OS : Arch Linux x86_64 
                             ​​ ​  Kernel : 6.8.2-arch2-1 
                             ​​ ​  Packages : 1007 (pacman), 6 (flatpak) 
                             ​​ ​  Shell : zsh 5.9 
                             ​​ ​  Resolution : 2560x1440 
                             ​​ ​  DE : Hyprland 
                             ​​ ​  Terminal : kitty 
                             ​​ ​  CPU : AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16) 
                             ​​ ​  GPU : NVIDIA GeForce RTX 4060 Max-Q / Mobile 
                             ​​ ​  GPU : AMD ATI 06:00.0 Phoenix1 
                             ​​ ​ ﬙ Memory : 5459MiB / 15171MiB

**Comment by kRHYME7** [comment]
Please confirm if you already tried this.

https://wiki.hyprland.org/Configuring/Multi-GPU/

**Reply by b1kr3m** [reply]
I have tried it, but after rebooting, it shows the sddm log in page only. I am not able to access the hyprland.

Could you please tell how I can revert the settings.
Because first time I tried it shows black screen after rebooting.....so I just reinstall the rice.

**Reply by Enigma1309** [reply]
press ctrl+alt+f3 (whatever function key doesn't really matter) 
login with your creds.
edit now you've got access to the system just undo the changes you have made that broke the system, (nano and vim/nvim might help you to edit files in tty)

**Reply by b1kr3m** [reply]
@kRHYME7 
I followed every instruction but hyprland does not load after change the config. 
please help me you find anything wrong......
my taking steps:
1. lspci | grep -E 'VGA|
         01:00.0 VGA compatible controller: NVIDIA Corporation AD107M [GeForce RTX 4060 Max-Q / Mobile] (rev a1)
06:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Phoenix1 (rev c2)

2.  ls -l /dev/dri/by-path
lrwxrwxrwx - root  5 Apr 15:06  pci-0000:01:00.0-card -> ../card1
lrwxrwxrwx - root  5 Apr 15:06  pci-0000:01:00.0-render -> ../renderD129
lrwxrwxrwx - root  5 Apr 15:06  pci-0000:06:00.0-card -> ../card0
lrwxrwxrwx - root  5 Apr 15:06  pci-0000:06:00.0-render -> ../renderD128

3. ln -sf /dev/dri/pci-0000:06:00.0-card ~/.config/hypr/card
❯ ls
 animations.conf
 card -> /dev/dri/pci-0000:06:00.0-card
 card -> /dev/dri/pci-0000:01:00.0-otherCard
 hyprdots-ctl
 hyprland.conf
 keybindings.conf
 monitors.conf
 nvidia.conf
 scripts

4.add in hyprland.conf
env = WLR_DRM_DEVICES,~/.config/hypr/card:~/.config/hypr/otherCard


dmesg:
[    5.030959] intel_rapl_common: Found RAPL domain package
[    5.030964] intel_rapl_common: Found RAPL domain core
[    5.032246] mt7921e 0000:03:00.0: WM Firmware Version: ____010000, Build Time: 20240219111038
[    5.047196] Generic FE-GE Realtek PHY r8169-0-200:00: attached PHY driver (mii_bus:phy_addr=r8169-0-200:00, irq=MAC)
[    5.054918] Bluetooth: hci0: AOSP extensions version v1.00
[    5.054924] Bluetooth: hci0: AOSP quality report is supported
[    5.055105] Bluetooth: MGMT ver 1.22
[    5.058768] NET: Registered PF_ALG protocol family
[    5.218025] r8169 0000:02:00.0 enp2s0: No native access to PCI extended config space, falling back to CSI
[    5.243917] r8169 0000:02:00.0 enp2s0: Link is Down
[    5.351378] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    5.860776] mt7921e 0000:03:00.0 wlo1: renamed from wlan0
[    5.967979] nvidia 0000:01:00.0: Enabling HDA controller
[    5.968101] snd_hda_intel 0000:01:00.1: enabling device (0000 -> 0002)
[    5.968311] snd_hda_intel 0000:01:00.1: Disabling MSI
[    5.968332] snd_hda_intel 0000:01:00.1: Handle vga_switcheroo audio client
[    5.968448] snd_hda_intel 0000:06:00.6: enabling device (0000 -> 0002)
[    5.985693] input: HDA NVidia HDMI/DP,pcm=3 as /devices/pci0000:00/0000:00:01.1/0000:01:00.1/sound/card0/input20
[    5.985778] input: HDA NVidia HDMI/DP,pcm=7 as /devices/pci0000:00/0000:00:01.1/0000:01:00.1/sound/card0/input21
[    5.985847] input: HDA NVidia HDMI/DP,pcm=8 as /devices/pci0000:00/0000:00:01.1/0000:01:00.1/sound/card0/input22
[    5.985911] input: HDA NVidia HDMI/DP,pcm=9 as /devices/pci0000:00/0000:00:01.1/0000:01:00.1/sound/card0/input23
[    5.990792] snd_hda_codec_realtek hdaudioC1D0: autoconfig for ALC257: line_outs=1 (0x14/0x0/0x0/0x0/0x0) type:speaker
[    5.990796] snd_hda_codec_realtek hdaudioC1D0:    speaker_outs=0 (0x0/0x0/0x0/0x0/0x0)
[    5.990798] snd_hda_codec_realtek hdaudioC1D0:    hp_outs=1 (0x21/0x0/0x0/0x0/0x0)
[    5.990800] snd_hda_codec_realtek hdaudioC1D0:    mono: mono_out=0x0
[    5.990801] snd_hda_codec_realtek hdaudioC1D0:    inputs:
[    5.990803] snd_hda_codec_realtek hdaudioC1D0:      Mic=0x19
[    5.990804] snd_hda_codec_realtek hdaudioC1D0:      Internal Mic=0x12
[    6.014502] ACPI Warning: \_SB.NPCF._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20230628/nsarguments-61)
[    6.014605] ACPI Warning: \_SB.PCI0.GPP0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20230628/nsarguments-61)
[    6.015036] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.GPP0.PEGP.GPS.NVD1], AE_NOT_FOUND (20230628/psargs-330)
[    6.015042] ACPI Error: Aborting method \_SB.PCI0.GPP0.PEGP.GPS due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
[    6.015047] ACPI Error: Aborting method \_SB.PCI0.GPP0.PEGP._DSM due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
[    6.036744] input: HD-Audio Generic Mic as /devices/pci0000:00/0000:00:08.1/0000:06:00.6/sound/card1/input24
[    6.036922] input: HD-Audio Generic Headphone as /devices/pci0000:00/0000:00:08.1/0000:06:00.6/sound/card1/input25
[    6.763188] nvidia-modeset: WARNING: GPU:0: Unable to read EDID for display device DP-2
[    6.772564] nvidia-modeset: WARNING: GPU:0: Unable to read EDID for display device DP-2
[    6.774331] [drm] Initialized nvidia-drm 0.0.0 20160202 for 0000:01:00.0 on minor 1
[    7.371941] usb 1-5.1: New USB device found, idVendor=048d, idProduct=c986, bcdDevice= 9.00
[    7.371955] usb 1-5.1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    7.371960] usb 1-5.1: Product: ITE Device(8176)
[    7.371965] usb 1-5.1: Manufacturer: ITE Tech. Inc.
[    7.426684] input: ITE Tech. Inc. ITE Device(8176) Keyboard as /devices/pci0000:00/0000:00:08.1/0000:06:00.3/usb1/1-5/1-5.1/1-5.1:1.0/0003:048D:C986.0005/input/input26
[    7.481773] input: ITE Tech. Inc. ITE Device(8176) Wireless Radio Control as /devices/pci0000:00/0000:00:08.1/0000:06:00.3/usb1/1-5/1-5.1/1-5.1:1.0/0003:048D:C986.0005/input/input27
[    7.482085] hid-generic 0003:048D:C986.0005: input,hiddev98,hidraw4: USB HID v1.10 Keyboard [ITE Tech. Inc. ITE Device(8176)] on usb-0000:06:00.3-5.1/input0
[    7.904085] r8169 0000:02:00.0 enp2s0: Link is Up - 1Gbps/Full - flow control off
[    8.820790] wlo1: authenticate with bc:22:28:3b:64:b1 (local address=a8:3b:76:1b:68:6f)
[    8.983473] wlo1: send auth to bc:22:28:3b:64:b1 (try 1/3)
[    8.987285] wlo1: authenticated
[    8.991129] wlo1: associate with bc:22:28:3b:64:b1 (try 1/3)
[    8.998007] wlo1: RX AssocResp from bc:22:28:3b:64:b1 (capab=0x411 status=0 aid=1)
![240405_15h47m23s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/97847882/2a084b17-bf1f-43ca-8fca-f34f81a38f34)
